src/Symfony/Component/Notifier/Bridge/Pushover/PushoverOptions.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Pushover;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+use Symfony\Component\Notifier\Notification\Notification;
+
+/**
+ * @author mocodo <https://github.com/mocodo>
+ *
+ * @see https://pushover.net/api
+ */
+final class PushoverOptions implements MessageOptionsInterface
+{
+    private const PRIORITIES = [-2, -1, 0, 1, 2];
+
+    private const SOUNDS = [
+        'pushover',
+        'bike',
+        'bugle',
+        'cashregister',
+        'classical',
+        'cosmic',
+        'falling',
+        'gamelan',
+        'incoming',
+        'intermission',
+        'magic',
+        'mechanical',
+        'pianobar',
+        'siren',
+        'spacealarm',
+        'tugboat',
+        'alien',
+        'climb',
+        'persistent',
+        'echo',
+        'vibrate',
+        'none',
+    ];
+
+    private array $options;
+
+    public function __construct(array $options = [])
+    {
+        $this->options = $options;
+    }
+
+    public static function fromNotification(Notification $notification): self
+    {
+        $options = new self();
+        $options->title($notification->getSubject());
+        $priority = match ($notification->getImportance()) {
+            Notification::IMPORTANCE_URGENT => 2,
+            Notification::IMPORTANCE_HIGH => 1,
+            Notification::IMPORTANCE_MEDIUM => 0,
+            Notification::IMPORTANCE_LOW => -1
+        };
+        $options->priority($priority);
+
+        return $options;
+    }
+
+    public function toArray(): array
+    {
+        $options = $this->options;
+        unset($options['attachment']);
+
+        return $options;
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return $this->options['device'] ?? null;
+    }
+
+    /**
+     * @see https://pushover.net/api#identifiers
+     */
+    public function device(?string $device): static
```

Comment:
why nullable?

Reply:
this one is on me, since Pushover allows to target one specific device or all, I though it could be interesting to reconfigure it to `null` if you instanciated the class with a specific device but it might be overkill, I removed the nullable as you suggested.